### PR TITLE
Update concierge components to add reschedule

### PR DESCRIPTION
**Pull request by unDemian**
**Built on top of https://github.com/Automattic/wp-calypso/pull/21364**

## Summary
This PR adds support for rescheduling concierge appointments. Rescheduling should be available at this url `/me/concierge/:siteSlug/:appointmentId/reschedule` . For more details please check out gh-hg-495

## Testing
1. Book an appointment by following the steps in https://github.com/Automattic/wp-calypso/pull/21364
2. Access the MC happiness/concierge page, under the `Lighthouse Only` tab you should see your appointment
3. Grab the id of your appointment, for now the only way to do that is by inspecting the element https://cloudup.com/c8dX8XZ2kOf and copying the `data-appointment-id` value
4. Access `http://calypso.localhost:3000/me/concierge/<business-site-slug>/<appointmentId>/reschedule`
5. The skeleton should reflect the page layout
6. The calendar page should reflect the rescheduling action
7. Select a timeslot and reschedule the appointment
8. The confirmation page should reflect the rescheduling
9. Using a non business site slug should show the upsell component
10. The MC page should reflect the new appointment date

### CODE REVIEW

File: client/me/concierge/reschedule/calendar-step.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	render() {
+		const { availableTimes, currentUserLocale, signupForm, site, translate } = this.props;
+
+		return (
+			<CalendarPage
+				actionText={ translate( 'Reschedule to this date' ) }
+				availableTimes={ availableTimes }
+				currentUserLocale={ currentUserLocale }
+				disabled={ signupForm.status === CONCIERGE_STATUS_BOOKING }
+				description={ translate( 'Please select a day to reschedule your Concierge session.' ) }
```
Comment (a8ci18n) [inline client/me/concierge/reschedule/calendar-step.js]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Please select a day to have your Concierge session.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/state/data-layer/wpcom/concierge/schedules/appointments/index.js [hunk 2]
```diff
@@ -39,6 +39,24 @@ export const bookConciergeAppointment = ( { dispatch }, action ) => {
...
+	dispatch( updateConciergeBookingStatus( CONCIERGE_STATUS_BOOKING ) );
+
+	dispatch(
+		http(
+			{
+				method: 'POST',
+				path: `/concierge/schedules/${ action.scheduleId }/appointments/${ action.appointmentId }/reschedule`,
+				apiNamespace: 'wpcom/v2',
+				body: {
+					begin_timestamp: action.beginTimestamp / 1000,
```
Comment (mattwondra) [inline client/state/data-layer/wpcom/concierge/schedules/appointments/index.js]: Maybe just leave a comment on this line similar to the one above in `toApi`: `// convert to UNIX timestamp.`

**Comment by mattwondra** [comment]
This branch has been rebased because the source branch was rebased, so you'll need to delete and refetch to keep in sync.

**Comment by unDemian** [comment]
Thanks for taking the time, I pushed some updates to address your feedback.

File: client/me/concierge/reschedule/confirmation-step.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+class ConfirmationStep extends Component {
+	render() {
+		const { site, translate } = this.props;
+
+		return (
+			<Confirmation
+				confirmationButton={ translate( 'Return to your dashboard' ) }
+				confirmationDescription={ translate(
+					'We will send you an email with information on how to get prepared.'
```
[the remark this answers is not in the document]
Comment (unDemian) [reply client/me/concierge/reschedule/confirmation-step.js]: Thanks, updated.

File: client/state/data-layer/wpcom/concierge/schedules/appointments/index.js (hunk 2 above)
Comment (unDemian) [reply client/state/data-layer/wpcom/concierge/schedules/appointments/index.js]: Right, updated.

File: client/me/concierge/reschedule/calendar-step.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		this.props.rescheduleConciergeAppointment(
+			WPCOM_CONCIERGE_SCHEDULE_ID,
+			appointmentId,
+			timestamp
+		);
+	};
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.signupForm.status === CONCIERGE_STATUS_BOOKED ) {
+			// go to confirmation page if booking was successfull
```
Comment (southp) [inline client/me/concierge/reschedule/calendar-step.js]: typo: successfull => successful :)

Comment (unDemian) [reply client/me/concierge/reschedule/calendar-step.js]: Nice catch, updated.

**Pull request merged**